### ISSUE TITLE
PG::ProtocolViolation error when upgrading to Rails 4.2.1

### ISSUE THREAD

**Issue by twojustin** [ErwinM/acts_as_tenant#114]
if i use version 0.3.6 and rails 4.2.1, there are no problems, but when i upgrade to version 0.3.7, i get the following error:

```
Failure/Error: delete :destroy, id: @clead.hash_id
     ActiveRecord::StatementInvalid:
       PG::ProtocolViolation: ERROR:  bind message supplies 4 parameters, but prepared statement "a76" requires 5
       : DELETE FROM "mail_merge_messages" WHERE "mail_merge_messages"."id" IN (SELECT "mail_merge_messages"."id" FROM "mail_merge_messages" INNER JOIN "campaign_leads" ON "campaign_leads"."id" = "mail_merge_messages"."campaign_lead_id" AND "campaign_leads"."company_id" = $1 WHERE "mail_merge_messages"."company_id" = $2 AND "campaign_leads"."company_id" = $3 AND "campaign_leads"."id" = $4 AND "mail_merge_messages"."state" = $5)
```

this error is triggered by the following code in rspec:

```rb
collection = CampaignLead.where(id: 2)
messages = MailMergeMessage.joins(:campaign_lead).merge(collection)
messages.draft.delete_all
```

`messages.draft` is a scope created by rails enum. 

```rb
class MailMergeMessage < ActiveRecord::Base
  acts_as_tenant :company
  belongs_to :campaign_lead

  enum state: {
    draft:        0
  }
end
```

**Comment by ErwinM** [comment]
@IgorDobryn This seems to be triggered by your PR? I don't understand why though, do you?

**Comment by IgorDobryn** [comment]
@ErwinM, I'm not sure. Didn't try it with 4.2.1. Still using it with 4.2.0.

**Comment by IgorDobryn** [comment]
@ErwinM, they released 4.2.3 recently. It includes fix for symbols passed to `ActiveRecord::Relation#select`.  Updating to 4.2.3 may fix this issue.

**Comment by twojustin** [comment]
the problem has been resolved.  I upgraded to 4.2.3, and changed my query from:

```ruby
Lead.where(CampaignLead.where('campaign_leads.campaign_id = ?', campaign_id)
                                            .where('campaign_leads.lead_id = leads.id')
                                            .select(1).exists.not)
```

to 

```rb
Lead.where(CampaignLead.unscoped.where('campaign_leads.campaign_id = ?', campaign_id)
                                            .where('campaign_leads.company_id = ?', @company_id)
                                            .where('campaign_leads.lead_id = leads.id')
                                            .select(1).exists.not)
```

at first i thought the problem was in AAT, but the heart of the problem lies in Rails is not doing proper parameter interpolation within the nested where statement.  Maybe this query is contorting activeRecord too much.  Either way.. problem resolved.  Thank you all for looking into it.
